supervisor/os/data_disk.py

```diff
@@ -23,21 +33,46 @@ def disk_used(self) -> Optional[Path]:
         """Return Path to used Disk for data."""
         return self.sys_dbus.agent.datadisk.current_device
 
+    @Job(conditions=[JobCondition.OS_AGENT])
+    async def load(self) -> None:
+        """Load DataDisk feature."""
+        # Update datadisk details on OS-Agent
+        if self.sys_dbus.agent.version >= AwesomeVersion("1.2.0"):
+            await self.sys_dbus.agent.datadisk.reload_device()
+
     @Job(
         conditions=[JobCondition.HAOS, JobCondition.OS_AGENT, JobCondition.HEALTHY],
         limit=JobExecutionLimit.ONCE,
         on_condition=HassOSJobError,
     )
     async def migrate_disk(self, new_disk: Path) -> None:
         """Move data partition to a new disk."""
-        # Need some error handling, but we need know what disk_used will return
+        # Validate integrity of the data input
```

Comment:
Should we abort here if new and current are the same path?

Reply:
it will, because we check if that is already a system used device

Reply:
but yeah, I think we need to improve that